binderhub/static/js/index.js

```diff
@@ -330,13 +326,28 @@ function loadingMain(providerSpec) {
     }
   }
   build(providerSpec, log, path, pathType);
+
+  // Looping through help text every few seconds
+  const launchMessageInterval = 6 * 1000
+  setInterval(nextHelpText, launchMessageInterval);
+
+  // If we have a long launch, add a class so we display a long launch msg
+  const launchTimeout = 120 * 1000
+  setTimeout(() => {
+    $('div#loader-links p.text-center').addClass("longLaunch");
+    nextHelpText();
+  }, launchTimeout)
+
   return false;
 }
 
 // export entrypoints
 window.loadingMain = loadingMain;
 window.indexMain = indexMain;
 
+// Show the logs by default
+log.show();
```

Comment:
Yay!

Reply:
Just tried this locally and the log doesn't open :-(

While pondering how to fix it I replaced the "building" state handler (L174) like so:
```
  image.onStateChange('building', function(oldState, newState, data) {
    $('#phase-building').removeClass('hidden');
    log.show();
  });
```
this works and only open the log if we are building the repo, otherwise it stays closed. It is a bit weird if it takes a moment for the build to start because the log is first closed (waiting for build to start...) and then pops open. If that is too weird I'd insert the `log.show()` in `image.onStateChange('waiting', function(oldState, newState, data) {` which I think would open the log for everyone.

Reply:
Hmmm - OK, well I assumed that would be an easy fix. If it isn't working and requires debugging, can we do this in a different PR? (I can just remove this change if so) I don't think it's strictly related to the image helper rotate thingy?

Reply:
Sounds good to me!

Reply:
ok, removed in latest push! lemme know if you want me to iterate on the messages or timing or anything else!